src/main/java/org/ccci/gto/persistence/DeadLockRetryAspectSupport.java

```diff
@@ -10,22 +10,33 @@
 import org.springframework.dao.ConcurrencyFailureException;
 
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import javax.persistence.EntityManager;
 import javax.persistence.OptimisticLockException;
 import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
 
-abstract class DeadLockRetryAspectSupport implements Ordered {
+public abstract class DeadLockRetryAspectSupport implements Ordered {
     private static final Logger LOG = LoggerFactory.getLogger(DeadLockRetryAspectSupport.class);
 
     @NotNull
-    @Autowired
     private EntityManagerResolver entityManagerResolver;
 
+    @Nonnull
+    private List<DeadLockDetector> deadLockDetectors = new ArrayList<>();
+
     private int defaultAttempts = 3;
 
     private int order = LOWEST_PRECEDENCE;
 
+    @Autowired(required = false)
+    public void setDeadLockDetectors(@Nullable final List<DeadLockDetector> detectors) {
+        deadLockDetectors = detectors != null ? detectors : new ArrayList<>();
+    }
+
+    @Autowired
```

Comment:
curious - was moving `@Autowired` from the field to this setter necessary? Or just a style change you wanted to make?

Reply:
wasn't necessary, I went through multiple iterations as I was working through the PR and I think it kind of just ended up on the Setter at the end of the iterations.